metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -168,13 +167,25 @@ func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 				},
 			}
 
+			if vm.Summary.Runtime.Host != nil {
+				event["host"] = vm.Summary.Runtime.Host.Value
+			} else {
+				logp.Debug("'Host', 'Runtime' or 'Summary' data not found. This is either a parsing error " +
+					"from vsphere library, an error trying to reach host/guest or incomplete information returned " +
+					"from host/guest", "")
+			}
+
 			// Get custom fields (attributes) values if get_custom_fields is true.
-			if m.GetCustomFields {
+			if m.GetCustomFields && vm.Summary.CustomValue != nil {
 				customFields := getCustomFields(vm.Summary.CustomValue, customFieldsMap)
 
 				if len(customFields) > 0 {
 					event["custom_fields"] = customFields
 				}
+			} else {
+				logp.Debug("Custom fields not activated or custom values not found/parse in Summary data. This " +
```

Comment:
See above comment for debug log

Reply:
Name of the module and metricset now added as beginning of debug message.